src/main/java/graphql/schema/PropertyDataFetcher.java

```diff
@@ -22,77 +18,13 @@ public PropertyDataFetcher(String propertyName) {
     public T get(DataFetchingEnvironment environment) {
         Object source = environment.getSource();
         if (source == null) return null;
-        if (source instanceof Map) {
-            return (T) ((Map<?, ?>) source).get(propertyName);
-        }
-        return (T) getPropertyViaGetter(source, environment.getFieldType());
```

Comment:
This optimisation is already present within `PropertyUtils` itself. See how the added test passes fine: https://github.com/graphql-java/graphql-java/pull/569/commits/f9eb924e7f36e929ba7299b4d9214b6c7fe3edff